.changesets/fix_geal_fix_hashing_algorithm.md

```diff
@@ -0,0 +1,26 @@
+### Fix cache key hashing algorithm ([Issue #5160](https://github.com/apollographql/router/issues/5160))
+
+> [!IMPORTANT]
+> If you have enabled [Distributed query plan caching](https://www.apollographql.com/docs/router/configuration/distributed-caching/#distributed-query-plan-caching), this release changes the hashing algorithm used for the cache keys.  On account of this, you should anticipate additional cache regeneration cost when updating between these versions while the new hashing algorithm comes into service.
+
+The Router contains a schema aware query hashing algorithm, designed to return the same hash across schema updates if they do not affect the query. It was used mainly to avoid planning a lot of queries when warming up the query plan cache. It was deactivated due to a regression. This reintroduces this algorithm, with a serie of reliability and performance fixes, along with better observability.
+
+The hashing algorithm can be activated in configuration, if query plan cache warm up is already enabled:
+
+```yaml title="router.yaml"
+supergraph:
+  query_planning:
+    warmed_up_queries: 100
+    experimental_reuse_query_plans: true
+```
```

Comment:
If removing the reference to warm-up, also remove the configuration.

```suggestion
```yaml title="router.yaml"
supergraph:
  query_planning:
    experimental_reuse_query_plans: true
```
```

Reply:
same as above, it is not usable without the warmup